src/orange/surf/VariantSurface.hh

```diff
@@ -18,5 +20,10 @@ namespace celeritas
 //! std::variant class of all surfaces.
 using VariantSurface = detail::VariantSurface_t;
 
+//---------------------------------------------------------------------------//
+// Apply a variant "daughter-to-parent" transform to a surface
+[[nodiscard]] VariantSurface
+apply_transform(VariantTransform const& left, VariantSurface const& right);
```

Comment:
Why `left` and `right` instead of `transform` and `surface`? Consistency with other `apply_transform` or copy-paste error?

Reply:
Consistency, but since it doesn't make sense to apply a surface to a transform I could rename them if you think it's clearer.

Reply:
I think it would be clearer to rename them, but I don't have a strong opinion